File: Contador/app/src/main/java/br/edu/scl/ifsp/ads/contador/MainActivity.java
package br.edu.scl.ifsp.ads.contador;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.text.Editable;
import android.text.TextWatcher;
import android.view.View;
import android.widget.AdapterView;
import android.widget.EditText;
import android.widget.RadioGroup;
import android.widget.TextView;
import android.widget.Button;

import br.edu.scl.ifsp.ads.contador.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {
    private ActivityMainBinding amb;

    private int contador = 0;
    private int add = 1;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        amb = ActivityMainBinding.inflate(getLayoutInflater());

        setContentView(amb.getRoot());

        amb.inicialSp.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
            @Override
            public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
                if(position == 0) {
                    add = 1;
                } else {
                    add = position * 5;
                }
            }

            @Override
            public void onNothingSelected(AdapterView<?> parent) {
                // NSA
            }
        });

        amb.cliqueBt.setOnClickListener(
                v -> amb.contadorTv.setText(String.valueOf(contador = contador + add))
        );
    }
}

File: SomeViewKotlin/app/src/main/java/br/edu/ifsp/scl/ads/someviewkotlin/MainActivity.kt
package br.edu.ifsp.scl.ads.someviewkotlin

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.view.View.OnClickListener
import android.widget.AdapterView
import android.widget.AdapterView.OnItemSelectedListener
import android.widget.Toast
import br.edu.ifsp.scl.ads.someviewkotlin.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    private lateinit var activityMainBinding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        activityMainBinding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(activityMainBinding.root)

        with(activityMainBinding) {
            estadoCivilSp.onItemSelectedListener = object: OnItemSelectedListener {
                override fun onItemSelected(
                    parent: AdapterView<*>?,
                    view: View?,
                    position: Int,
                    id: Long
                ) {
                    if (position == 2) { // Casado
                        conjugeLl.visibility = View.VISIBLE
                    } else {
                        conjugeLl.visibility = View.GONE
                        nomeConjugeEt.setText("")
                        sobrenomeConjugeEt.setText("")
                    }
                }

                override fun onNothingSelected(parent: AdapterView<*>?) {
                    // NSA
                }
            }

            conjugeLl.visibility = View.GONE
        }

        activityMainBinding.salvarBt.setOnClickListener {
                Toast.makeText(this, "Foi clicado", Toast.LENGTH_LONG).show()
        }
    }
}